src/os/newstore/NewStore.cc

```diff
@@ -3280,10 +3306,12 @@ int NewStore::_do_write_all_overlays(TransContext *txc,
     op->offset = p->first;
     op->length = p->second.length;
     op->fid = f.fid;
+    // The overlays will be removed from the db after applying the WAL
+    op->data_in_overlay = true;
+    op->nid = o->onode.nid;
+    op->overlays.push_back(p->second);
     op->data.substr_of(bl, p->second.value_offset, p->second.length);
```

Comment:
seems we can drop the data copy in  line 3283? The op->data will not be encoded at all.